dashboard/app/views/levels/_level_group.haml

```diff
@@ -6,21 +6,38 @@
 
 = render partial: 'levels/common_audio'
 
-- data["levels"].each_with_index do |level_name, index|
-  - level_blob = Level.find_by_name(level_name)
-  - level = level_blob
-  - # Parse the array of the level resuts; see sample below in JavaScript.
+- page = params[:puzzle_page] ? (params[:puzzle_page].to_i - 1) : 0
+
+- data["pages"][page]["levels"].each_with_index do |level_name, index|
+  - level = Level.find_by_name(level_name)
   - if @last_attempt
-    - last_attempt = JSON.parse(@last_attempt).find{|item| item["level_id"]==level.id}["result"]
+    - last_attempt_entry = JSON.parse(@last_attempt).find{|item| item["level_id"]==level.id}
+    - last_attempt = last_attempt_entry["result"] if last_attempt_entry
   - else
     - last_attempt = ""
-  = render partial: 'levels/single_multi', locals: {index: index, standalone: false, level: level, last_attempt: last_attempt }
+  = render partial: 'levels/single_multi', locals: {page: page, index: index, standalone: false, level: level, last_attempt: last_attempt }
 
 = render partial: 'levels/common_audio'
-= render partial: 'levels/dialog', locals: {app: app, data: data}
+
+- if page < data["pages"].length - 1
+  %a.btn.btn-large.btn-primary.nextPageButton
+    =t('nextPage')
+- else
+  = render partial: 'levels/dialog', locals: {app: app, data: data}
+  %a.btn.btn-large.btn-primary.unsubmitButton{style: "display: none"}
+    =t('unsubmit')
 
 :javascript
   #{render template: 'levels/multi.js', formats: [:js]}
 
-  var levelCount = #{data["levels"].count};
+  var levelCount = #{data["pages"][page].length};
```

Comment:
hm, I would maybe feel better if these were function arguments instead of globals (like the way we pass things to dashboard.buildHeader)